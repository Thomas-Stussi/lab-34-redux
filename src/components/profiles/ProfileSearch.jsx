import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchUser, setSearch } from '../../actions/githubActions';

const ProfileSearch = () => {
  const dispatch = useDispatch();
  const userName = useSelector(state => state.search);

  const handleSubmit = () => {
    dispatch(fetchUser(userName));
  };

  const handleChange = ({ target }) => {
    dispatch(setSearch(target.value));
  };

  return (
    <>
      <input onChange={handleChange} />
      <button onClick={handleSubmit}>Search Github Profiles</button>
    </>
  );
};

export default ProfileSearch;
