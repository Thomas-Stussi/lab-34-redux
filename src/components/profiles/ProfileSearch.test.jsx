import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ProfileSearch from './ProfileSearch';
import { Provider } from 'react-redux';
import store from '../../store';

describe('ProfileSearch component', () => {
  afterEach(() => cleanup());
  it('renders ProfileSearch', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <ProfileSearch />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
