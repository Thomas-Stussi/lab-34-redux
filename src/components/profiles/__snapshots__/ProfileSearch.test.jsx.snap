// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileSearch component renders ProfileSearch 1`] = `
<DocumentFragment>
  <input />
  <button>
    Search Github Profiles
  </button>
</DocumentFragment>
`;
